solidity/contracts/BancorNetwork.sol

```diff
@@ -456,19 +414,8 @@ contract BancorNetwork is TokenHolder, ContractRegistryClient, ReentrancyGuard {
         }
 
         IERC20 targetToken = stepData.targetToken;
-
-        // ETH / EtherToken
-        if (etherTokens[targetToken]) {
-            // newer converter should send ETH directly to the beneficiary
-            assert(!stepData.isV28OrHigherConverter);
-
-            // EtherToken converter - withdraw the ETH and transfer to the beneficiary
-            IEtherToken(address(targetToken)).withdrawTo(_beneficiary, _amount);
-        }
-        // other ERC20 token
-        else {
-            targetToken.safeTransfer(_beneficiary, _amount);
-        }
+        assert(targetToken != ETH_RESERVE_ADDRESS);
```

Comment:
Perhaps a `require`?

Reply:
If this condition is not met, then it's because of a bug in the contract, not because of a wrong user input.
So IMO, `assert` is appropriate here, while `require` is not.